### PR TITLE
🎨 integrate document type message on chats list

### LINKED ISSUE #836: 🎨 Integrate Document Type Message On Chats List
## User stories

As a user, I want to view the document-type message when it's shared with me or any of my users on the chats page and have an option to download it.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

#### Note: This issue is dependent on #835.
---

### Technical Description
The current design is: 
Missing Message

It should be changed to:
New Document Message

The chats can be found at `libs\features\convs-mgr\conversations\chats`. 
The messages can be found at `libs\features\convs-mgr\conversations\messaging`

[Design Link](https://xd.adobe.com/view/a9bd70b9-0f03-4663-adf4-27eba09af458-fcee/)

### Acceptance Criteria

1. I should view a document-type message whether it's from the bot or the user.
3. All changes should be isolated to the following  `libs\features\convs-mgr\conversations\chats` or `libs\features\convs-mgr\conversations\messaging` if necessary.

### PR DESCRIPTION
**Pull request by darlene9497**
# Description

This pull request addresses the issue of integrating the document message on the chats list.

Fixes #836 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Screenshot (optional)
Screenshot from 2023-11-27 08-15-25

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: libs/features/convs-mgr/conversations/messaging/src/lib/components/chat-message/chat-message.component.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import {Component, OnInit, Input, AfterViewInit } from '@angular/core';
 import { DomSanitizer } from '@angular/platform-browser';
-import { Message, MessageDirection, TextMessage } from '@app/model/convs-mgr/conversations/messages';
+import { HttpClient } from '@angular/common/http';
+import { Message, MessageDirection, TextMessage, DocumentMessage } from '@app/model/convs-mgr/conversations/messages';
+import { DocumentMessageBlock } from '@app/model/convs-mgr/stories/blocks/messaging';
+
+import { ICONS_AND_TITLES } from '../../../../../../stories/blocks/library/main/src/lib/assets/icons-and-titles';
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/conversations/messaging/src/lib/components/chat-message/chat-message.component.ts:7]: This import is too relative. Export the file from it's module so you can use the aliased path to import it.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Very good attempt

File: libs/features/convs-mgr/conversations/messaging/src/lib/components/chat-message/chat-message.component.html [hunk 2]
```diff
@@ -20,6 +20,19 @@
...
+          <div class="icon-and-size">
+            <img [src]="svgIcon" class="svgIcon" alt="">
+            <div class="file-size">{{ fileSize }} KB</div>
+          </div>
+          <div class="link-download">
+            <a [href]="documentUrl" target="_blank">{{ documentName }}</a>
+            <img src="assets/icons/download.svg" class="download-btn" alt="" (click)="downloadDocument()">
+          </div>
+        </div>
+      </div>
```
[the remark this answers is not in the document]
Comment (darlene9497) [reply libs/features/convs-mgr/conversations/messaging/src/lib/components/chat-message/chat-message.component.html:34]: @LemmyMwaura I have done this but passing it as app-resource-message, only the icons are displaying. Kindly advise

[the remark this answers is not in the document]
Comment (LemmyMwaura) [reply libs/features/convs-mgr/conversations/messaging/src/lib/components/chat-message/chat-message.component.html]: Okay, i'll test your branch tomorrow then give you feedback.

**Pull request closed without merging**